library/protocol.cpp

```diff
@@ -5,513 +5,604 @@
 #include "rapidjson/document.h"
 
 namespace msgpack {
+MSGPACK_API_VERSION_NAMESPACE(MSGPACK_DEFAULT_API_NS) {
+namespace adaptor {
 using namespace ioremap::elliptics;
 
-inline dnet_time &operator >>(msgpack::object o, dnet_time &v) {
-	if (o.type != msgpack::type::ARRAY || o.via.array.size != 2) {
-		throw msgpack::type_error();
-	}
-
-	object *p = o.via.array.ptr;
-	p[0].convert(&v.tsec);
-	p[1].convert(&v.tnsec);
-	return v;
-}
-
-template <typename Stream>
-inline msgpack::packer<Stream> &operator <<(msgpack::packer<Stream> &o, const dnet_time &v) {
-	o.pack_array(2);
-	o.pack_fix_uint64(v.tsec);
-	o.pack_fix_uint64(v.tnsec);
-	return o;
-}
+template<>
+struct convert<dnet_time> {
+	msgpack::object const& operator()(msgpack::object const& o, dnet_time& v) const {
+		if (o.type!=msgpack::type::ARRAY || o.via.array.size != 2) {
+			throw msgpack::type_error();
+		}
 
-inline dnet_read_request &operator >>(msgpack::object o, dnet_read_request &v) {
-	if (o.type != msgpack::type::ARRAY || o.via.array.size < 4) {
-		throw msgpack::type_error();
+		auto p = o.via.array.ptr;
+		p[0].convert(v.tsec);
+		p[1].convert(v.tnsec);
```

Comment:
JFMI: Before commit here was `&v.tnsec` - what a reason to change?

Reply:
msgpack API changes, now it accepts references instead of pointers.